mega-sdist-0.3.3.1

```
mega-sdist.hs:257:23: error:
    Ambiguous occurrence ‘Header’
    It could refer to either ‘Network.HTTP.Simple.Header’,
                             imported from ‘Network.HTTP.Simple’ at mega-sdist.hs:13:1-26
                             (and originally defined in ‘http-types-0.12.2:Network.HTTP.Types.Header’)
                          or ‘Data.Conduit.Tar.Header’,
                             imported from ‘Data.Conduit.Tar’ at mega-sdist.hs:14:1-23
                             (and originally defined in ‘Data.Conduit.Tar.Types’)
    |
257 |                    -> Header
    |                       ^^^^^^
```

CC @snoyberg. Going to drop `mega-sdist` for now.

--- a/build-constraints.yaml
+++ b/build-constraints.yaml
@@ -411,7 +411,7 @@ packages:
 
     "Michael Snoyman michael@snoyman.com @snoyberg":
         - bzlib-conduit
-        - mega-sdist < 0 # https://github.com/commercialhaskell/stackage/issues/4215
+        - mega-sdist
         - case-insensitive
         - classy-prelude-yesod
         - conduit-combinators

